test/configuration_test.dart

```diff
@@ -153,4 +153,48 @@ Future<void> main([List<String>? args]) async {
     realm1.close();
     realm2.close();
   });
+  
+  test('Configuration - disableFormatUpgrade=true throws error', () async {
+    final realmBundleFile = "test/data/realm_files/realm-bundle.realm";
+    final realmDir = "test/disableFormatUpgradeTrue";
+    final realmPath = combineFilePath(realmDir, Configuration.defaultPath);
+
+    await Directory(realmDir).create();
+    var config = Configuration([Car.schema])..path = realmPath;
+    var realm = Realm(config);
+    realm.close();
+
+    Realm.deleteRealm(realmPath);
+    var file = File(realmBundleFile);
+    await file.copy(realmPath);
+
+    config = Configuration([Car.schema], disableFormatUpgrade: true)..path = realmPath;
+    expect(() {
+      realm = Realm(config);
+    }, throws<RealmException>("The Realm file format must be allowed to be upgraded in order to proceed"));
+    realm.close();
+
+    await Directory(realmDir).delete(recursive: true);
+  }, skip: isFlutterPlatform);
+
+  test('Configuration - disableFormatUpgrade=false', () async {
+    final realmBundleFile = "test/data/realm_files/realm-bundle.realm";
```

Comment:
Can't comment on the right thing, but `realm-bundle.realm` is an overly vague name. We'll likely need to add more such files, e.g. files for migration, etc. so it's important to name them in a more descriptive way. This one could be `old-format.realm`.

Reply:
Done.